### PR TITLE
Improve the forgotten password flow

### PR DESCRIPTION
**Pull request by ErisDS**
- if the user has filled out their email already when they hit 'forgotten password' then automatically  fill out and submit the form
- this will look better when we have spinners on the buttons

### CODE REVIEW

**Comment by ErisDS** [comment]
The idea here is to simply minimise the steps required to reset a password as there is an unnecessary one at present. Would be nice if we could do it without flashing up the forgotten password form, but I don't thing it's a blocker. 

When we get #3928 done, the action will be more clearly in progress. 

Would love some ember feedback if there's a better way to do this though :grin:

File: core/client/app/controllers/signin.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import ValidationEngine from 'ghost/mixins/validation-engine';
 
 var SigninController = Ember.Controller.extend(SimpleAuth.AuthenticationControllerMixin, ValidationEngine, {
     authenticator: 'simple-auth-authenticator:oauth2-password-grant',
+    needs: ['forgotten'],
```
Comment (novaugust) [inline core/client/app/controllers/signin.js]: the `needs` api is going the way of the dodo - mind switching to the new [Ember.inject.controller](http://emberjs.com/api/classes/Ember.inject.html#method_controller)?

**Comment by ErisDS** [comment]
@novaugust updated to use the new Ember.inject thingamabob :)

**Comment by jaswilli** [comment]
This looks good. Re: the reset form flashing, does the transition to the "forgotten" route even need to happen if the email has been filled in?

**Comment by ErisDS** [comment]
@jaswilli no not really, I was just using that to do this in a simple and DRY way, as I wasn't sure how else it could be done? Do mixins work for actions? I'm still pretty green on the ember stuff - I stab around in the dark and mostly it works :grin:

**Comment by jaswilli** [comment]
I wasn't thinking anything more clever than literally just not doing the transition by way of `return`ing after sending `doForgotten` to the controller if we already have the email. It seems like it shouldn't be necessary in that case.

**Comment by ErisDS** [comment]
Ah - I didn't realise returning would work that way - however, having just tested it out, it works but simply stays on the signin screen with no notification.

**Comment by jaswilli** [comment]
I suspect that's because the notification is set as "delayed" which means that it doesn't get triggered until a route transition occurs. We could pass a parameter to `doForgotten` to make the notification not delayed and I suspect that would work.

**Comment by ErisDS** [comment]
@jaswilli holy beep you're totally right. Man that's beautiful. Thanks for the tips :) PR updated. @JohnONolan you're gonna flippin' :heart: this.

**Pull request merged**